circuitpython/network/golemio_odjezdy_mhd/code.py

```diff
@@ -0,0 +1,132 @@
+"""
+What about using some open data that are provided by your government or city?
+In this example we will connect to Prague data platform Golemio.cz and display departure times at some public transport stop.
+
+1. To get started, let's set up an API Golemio account, and generate your API key for free at https://api.golemio.cz/api-keys/auth/sign-up Don't forget to set the api key as an environment variable  GOLEMIO_API_KEY in the settings.toml file.
+2. Next, set the WiFi SSID and password as environment variables CIRCUITPY_WIFI_SSID and CIRCUITPY_WIFI_PASSWORD in the settings.toml file.
+3. Finally set stop STOP_NAME or STOP_GTFS_ID in settings.toml file for stop that you want you inspect.
+   - STOP_NAME is case-sensitive fullname of stop, MUST BE written with diacritics, no acronyms are accepted.
+   - GTFS ID (General Transit Feed Specification ID) is unique id for every stop, where you can be more selective,
+     eg. U400Z101P is GTFS ID for Muzeum metro A direction Depo Hostivar.
+     List of GTFS ID can be found at https://data.pid.cz/stops/json/stops.json
+
+If you are interested than you might check departure times at https://mapa.pid.cz/ they should be exatly same like in this script.
+
+You can find the required libraries in the CircuitPython library bundle (https://circuitpython.org/libraries).
+"""
+import os
+import wifi
+import ssl
+import socketpool
+import time
+import zlib
+import json
+import adafruit_requests
+
+
+# If you have font with Czech diacritics than you will not need this
+def removeCzechDiacritics(text):
+    """
+    Convert Czech characters to plain ASII.
+    """
+    diacritics    = "ÁÉĚÍÓÚŮÝČĎŇŘŠŤŽáéěíóúůýčďňřšťž"
+    no_diacritics = "AEEIOUUYCDNRSTZaeeiouuycdnrstz"
+    
+    result = ""
+    
+    for char in text:
+        if char in diacritics:
+            index = diacritics.index(char)
+            result += no_diacritics[index]
+        else:
+            result += char
+    
+    return result
+
+
+def PrintStopInfo(stop):
+    print(f"Departures from {removeCzechDiacritics(stop["stop_name"]).upper()}")
+
+def PrintDepartures(departures):
+    for depart in departures:
+        if depart["trip"]["is_air_conditioned"]:
+            aircondition = "*"
+        else:
+            aircondition = " "
+
+        print(f"{depart["route"]["short_name"]:>4} {aircondition} {removeCzechDiacritics(depart["trip"]["headsign"]):25} {depart["departure_timestamp"]["minutes"]:>3} min")
+
+def PrintInfotexts(infotexts):
+    for inf in infotexts:
+        print(inf["text"])
+
+
+def ConnectWifi():
+    wifi.radio.connect(os.getenv("CIRCUITPY_WIFI_SSID"), os.getenv("CIRCUITPY_WIFI_PASSWORD"))
+    print("Connected to WiFi", os.getenv("CIRCUITPY_WIFI_SSID"))
+
+    pool = socketpool.SocketPool(wifi.radio)
+    request = adafruit_requests.Session(pool, ssl.create_default_context())
+    return request
+
+
+def AskGolem(request):
+
+    if os.getenv("GOLEMIO_API_KEY") is None or len(os.getenv("GOLEMIO_API_KEY")) < 80:
```

Comment:
ENV should be loaded on the start of application and then passed on as function arguments